src/parser/HttpParser.ts

```diff
@@ -64,20 +70,26 @@ export class HttpParser {
     };
     // Check if mock file exists
     if (fs.existsSync(mockFile)) {
-      this.prepareResponse(mockFile);
+      response = await this.prepareResponse(mockFile);
     } else {
       logger.error(`No suitable mock file found: ${mockFile}`);
       if (fs.existsSync(path.join(this.mockDir, "__", "GET.mock"))) {
-        logger.debug(`Found a custom global override for default response. Sending custom default response.`);
-        this.prepareResponse(path.join(this.mockDir, "__", "GET.mock"));
+        logger.debug(
+          `Found a custom global override for default response. Sending custom default response.`
+        );
+        response = await this.prepareResponse(
+          path.join(this.mockDir, "__", "GET.mock")
+        );
       } else {
         //If no mockFile is found, return default response
-        logger.debug(`No custom global override for default response. Sending default Camouflage response.`);
-        this.res.statusCode = response.status;
-        this.res.set(response.headers)
-        this.res.send(response.body);
+        logger.debug(
+          `No custom global override for default response. Sending default Camouflage response.`
+        );
       }
     }
+
+    response.status = parseFloat((<unknown>response.status) as string);
```

Comment:
This converts the status code from string to number
```
export interface HttpParserResponse {
  status: number;
```

So in that sense it's not redundant.

Reply:
I think we have covered all the assignments:
1. The response has been initialised  with `status: 404,` a number.
2. Any assignments to `response.status` in `prepareResponse` method, takes care of the conversion from string to number. `response.status = <number>(<unknown>line.match(regex)[1]);`

Apart from these two cases, I am not sure there would be any other use cases where the conversion at L91 makes sense, however we could leave it there to catch any unhandled scenarios (e.g. L187 `response.status = codeResponse["status"];`)

Reply:
👍